plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/ExternalIssue.java

```diff
@@ -53,4 +54,11 @@ public interface ExternalIssue extends IIssue {
    */
   RuleType type();
 
+  /**
+   * Characteristic of the issue according to Clean Code Taxonomy.
+   * @since 9.16
+   */
+  @CheckForNull
+  RuleCharacteristic characteristic();
```

Comment:
Would it makes sense to annotate this method with @Beta as we will need to release it and will potentially still be subject to change?

Reply:
Anyone else could comment on this? I think this makes sense but we don't use @Beta too often. Not sure.

Reply:
In principle, I think this makes sense (and adding it to related methods and the `RuleCharacteristic` class as well).